[Feature #156881920] Implement signup with social media and profile picture upload

nzediegwu1 opened the pull request:
#### What does this PR do?
Implement signup with social media and profile picture upload upon signup

#### Description of Task to be completed?
1. Upgrade mongoose installation
2. Modify user model to include picture and publicId fields
3. Create profilePicture angular controller to handle signup with picture upload
4. Implement signup with facebook (not yet tested)
5. Implement signup with google
6. Implement signup with twitter
7. Implement signup with github
8. Add custom style to signup form
9. Correct houndci spacing error
10. Delete duplicate .env.sample
11. Resolve local conflicts from latest develop branch

#### How should this be manually tested?
On terminal, run
1. `npm start`
2. Navigate to the app landing page and click signup
3. Signup using facebook, then logout and
4. Signup using github, then logout and
5. Signup using twitter, then logout and
6. Signup using google, then logout and
7. Upload signup by by supplying name, email and password, with profile picture uploaded

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#156881920](https://www.pivotaltracker.com/story/show/156881920)

#### Screenshots (if appropriate)
None
#### Questions:
None

Commit:
[Feature #156881920] Implement signup with social media and profile picture

1. Upgrade mongoose installation
2. Modify user model to include picture and publicId fields
3. Create profilePicture angular controller to handle signup with picture upload
4. Implement signup with facebook (not yet tested)
5. Implement signup with google
6. Implement signup with twitter
7. Implement signup with github
8. Add custom style to signup form
9. Correct houndci spacing error
10. Delete duplicate .env.sample
11. Resolve local conflicts from latest develop branch
[Finishes #156881920]

nzediegwu1 committed:
Merge branch 'develop' into ft-signup-with-profile-image-156881920

Chieze-Franklin reviewed (changes requested):
This PR contains sensitive info that should not be committed to Github. Please fix!

Commit:
[Fix #156881920 ] Refactor code

1.  Move sensitive info into .env file
2. Correct synthax errors breaking app
3. Resolve eslint errors
[Finishes #156881920 ]

Commit:
Merge branch 'ft-signup-with-profile-image-156881920' of https://github.com/andela/dark-knight-cfh into ft-signup-with-profile-image-156881920
